devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1319,20 +1338,72 @@ bool HumanStateProvider::impl::applyRpcCommand()
+        linkToCalibrateIndices.resize(kinDynComputations->getNrOfLinks());
+        std::iota(linkToCalibrateIndices.begin(), linkToCalibrateIndices.end(), 0);
+
+        // add all the joints of the model to [jointZeroIndices]
+        jointZeroIndices.resize(kinDynComputations->getNrOfDegreesOfFreedom());
+        std::iota(jointZeroIndices.begin(), jointZeroIndices.end(), 0);
 
         // Compute secondary calibration for the selected links setting to zero the given joints
-        computeSecondaryCalibrationRotations(jointZeroIndices, linkToCalibrateIndices);
-        return true;
+        computeSecondaryCalibrationRotationsForChain(jointZeroIndices, baseTransformSolution, linkToCalibrateIndices);
+        break;
+    }
+    case rpcCommand::calibrate: {
+        // Select the joints to be set to zero and the link to be add the secondary calibration
+        selectChainJointsAndLinksForSecondaryCalibration(linkName, childLinkName, jointZeroIndices, linkToCalibrateIndices);
+        // Compute secondary calibration for the selected links setting to zero the given joints
+        computeSecondaryCalibrationRotationsForChain(jointZeroIndices, baseTransformSolution, linkToCalibrateIndices);
+        break;
+    }
+    case rpcCommand::calibrateSubTree: {
+        // Select the joints to be set to zero and the link to be add the secondary calibration
+        selectChainJointsAndLinksForSecondaryCalibration(linkName, childLinkName, jointZeroIndices, linkToCalibrateIndices);
+        // Compute secondary calibration for the selected links setting to zero the given joints
+        computeSecondaryCalibrationRotationsForChain(jointZeroIndices, baseTransformSolution, linkToCalibrateIndices);
+    }
+    case rpcCommand::calibrateRelativeLink: {
+        eraseSecondaryCalibration(childLinkName);
+        // Compute the relative transform at zero configuration
+        // setting to zero all the joints
+        iDynTree::VectorDynSize jointPos;
+        jointPos.resize(jointConfigurationSolution.size());
+        jointPos.zero();
+        kinDynComputations->setJointPos(jointPos);
+        iDynTree::Rotation relativeRotationZero = kinDynComputations->getWorldTransform(linkName).getRotation().inverse() * kinDynComputations->getWorldTransform(childLinkName).getRotation();
+        secondaryCalibrationRotation = linkTransformMatricesRaw.at(childLinkName).getRotation().inverse() * linkTransformMatrices.at(linkName).getRotation() * relativeRotationZero;
+        secondaryCalibrationRotations.emplace(childLinkName,secondaryCalibrationRotation);
+        yInfo() << LogPrefix << "secondary calibration for " << childLinkName << " is set";
+        break;
+     }
+    case rpcCommand::setRotationOffset: {
+        eraseSecondaryCalibration(linkName);
+        secondaryCalibrationRotation = iDynTree::Rotation::RPY( 180 * commandPro->roll / 3.14 , 180 * commandPro->pitch / 3.14 , 180 * commandPro->yaw / 3.14 );
```

Comment:
In the help it is written  `response.addString("Enter <setRotationOffset <linkName> <r p y [deg]>>` if the values are in degree then why we are changing them to degree again.  I checked here https://robotology.github.io/idyntree/master/classiDynTree_1_1Rotation.html#a8a09f91f852dfc6eef646cdf737cb572 , it is not written if it gets the values in degree or radians, but I assume it should be in radians!

Reply:
I have corrected the conversion in order to transform degrees to radians in https://github.com/robotology/human-dynamics-estimation/pull/211/commits/dfa4a044dc57cb98881563e936c104e9fb38d24c.

I would leave the input values as degrees because it is easier to evaluate them easily to correct the offset.